oxd-server/src/main/java/org/gluu/oxd/server/service/DiscoveryService.java

```diff
@@ -99,7 +136,33 @@ public UmaMetadata getUmaDiscoveryByOxdId(String oxdId) {
         validationService.notBlankOxdId(oxdId);
 
         Rp rp = rpSyncService.getRp(oxdId);
-        return getUmaDiscovery(rp.getOpHost(), rp.getOpDiscoveryPath());
+        return getUmaDiscovery(rp.getOpConfigurationEndpoint(), rp.getOpHost(), rp.getOpDiscoveryPath());
+    }
+
+    public UmaMetadata getUmaDiscovery(String opConfigurationEndpoint, String opHost, String opDiscoveryPath) {
```

Comment:
Same story for UMA, please refactor existing method and re-use.

Reply:
Done